src/array.jl

```diff
@@ -546,7 +546,65 @@ Adapt.adapt_storage(::CuArrayAdaptor{B}, xs::AbstractArray{T,N}) where {T<:Compl
 Adapt.adapt_storage(::CuArrayAdaptor{B}, xs::AbstractArray{T,N}) where {T<:Union{Float16,BFloat16},N,B} =
   isbits(xs) ? xs : CuArray{T,N,B}(xs)
 
+"""
+    cu(A::AbstractArray{T}; unified=false)
+
+Opinionated GPU array adaptor, which may alter the element type:
+* For `T<:AbstractFloat`, it makes a `CuArray{Float32}` for performance reasons.
+  (Except that `Float16` and `BFloat16` element types are not changed.)
+* For `T<:Complex{<:AbstractFloat}` it makes a `CuArray{ComplexF32}`.
+* For other `isbitstype(T)`, it makes a `CuArray{T}`.
+
+By contrast, `CuArray(A)` never changes the element type.
+
+Uses `adapt(CuArrayAdaptor(), A)`, which works through some wrapper structs
```

Comment:
I'm not sure it's valuable to point to CuArrayAdaptor, that seems like an implementation detail. Just mentioning it recurses into types (using Adapt) seems sufficient to me?

Reply:
Ok, see if you like the latest version.